larq_zoo/model_factory.py

```diff
@@ -0,0 +1,45 @@
+from typing import Callable, Optional, Tuple, Union
+
+import tensorflow as tf
+from zookeeper import ComponentField, Field
+from zookeeper.tf import Dataset
+
+from larq_zoo import utils
+
+# FIXME
+QuantizerType = Union[
+    tf.keras.layers.Layer, Callable[[tf.Tensor], tf.Tensor], str, None
+]
+ContraintType = Union[tf.keras.constraints.Constraint, str, None]
+DimType = Optional[int]
+
+
+class ModelFactory:
```

Comment:
Perhaps this class definition belongs in Zookeeper rather than Zoo? It seems to me like it would be useful to any Zookeeper consumer, and in larq/zookeeper#110 I deleted the existing built-in `Model` class, so this could replace it. Thoughts @ reviewers?

Reply:
Yes, as long as the number of classes is optional when you don't include the top, this seems a useful class to have :+1:

Reply:
Yes, I think this would be useful to have in zookeeper.

Reply:
I've discussed this with @lgeiger; we're going to leave it in Zoo for now (can always be imported from Zoo) but can transition it to Zookeeper later on.